packages/sanity/src/core/studio/StudioThemeProvider.tsx

```diff
@@ -10,11 +12,14 @@ interface StudioThemeProviderProps {
 /** @internal */
 export function StudioThemeProvider({children}: StudioThemeProviderProps) {
   const theme = useActiveWorkspace().activeWorkspace.theme
-  const {scheme} = useColorScheme()
+  const colorScheme = useColorScheme()
+  const scheme = theme.__legacy ? (theme.__dark ? 'dark' : 'light') : colorScheme.scheme
 
   return (
-    <ThemeProvider scheme={scheme} theme={theme} tone="transparent">
-      <LayerProvider>{children}</LayerProvider>
-    </ThemeProvider>
+    <ColorSchemeContext.Provider value={{...colorScheme, scheme}}>
```

Comment:
Should this be memoized? I guess there aren't too many things outside of the theme provider that causes rerenders, but when rendered inside of other apps it might be?